libraries/Ethernet/examples/udpntpclient_pedantic/UdpNtpClient_Pedantic.ino

```diff
@@ -0,0 +1,160 @@
+/*
+
+ Udp NTP Client
+
+ Get the time from a Network Time Protocol (NTP) time server
+ Demonstrates use of UDP sendPacket and ReceivePacket
+ For more on NTP time servers and the messages needed to communicate with them,
+ see http://en.wikipedia.org/wiki/Network_Time_Protocol
+
+ created 4 Sep 2010
+ by Michael Margolis
+ modified 9 Apr 2012
+ by Tom Igoe
+
+ This code is in the public domain.
+
+ Modified by David Henry to show where all the 'magic numbers' come from.
+ You need to read the RFC-1305 spec to understand https://tools.ietf.org/html/rfc1305
+ mgadriver@gmail.com
+
+*/
+
+#include <SPI.h>
+#include <Ethernet.h>
+#include <EthernetUdp.h>
+#include "RFC1305.h"
+
+// Enter a MAC address for your controller below.
+// Newer Ethernet shields have a MAC address printed on a sticker on the shield
+byte mac[] = {
+  0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED
+};
+
+unsigned int localPort = 8888;       // local port to listen for UDP packets
+
+char timeServer[] = "time.nist.gov"; // time.nist.gov NTP server
+
+#define NTP_PACKET_SIZE sizeof(struct sRFC1305)
+
+struct sRFC1305 packetBuffer; //buffer to hold incoming and outgoing packets
+
+// A UDP instance to let us send and receive packets over UDP
+EthernetUDP Udp;
+
+void setup()
+{
+  // Open serial communications and wait for port to open:
+  Serial.begin(9600);
+  while (!Serial) {
+    ; // wait for serial port to connect. Needed for Leonardo only
+  }
+  //Serial.println(NTP_PACKET_SIZE); // just for debugging
+  //Serial.println(ENDIAN_SWAP_32(0x11223344),HEX);
+  //Serial.println(ENDIAN_SWAP_16(0xAABB),HEX);
+  // start Ethernet and UDP
+  if (Ethernet.begin(mac) == 0) {
```

Comment:
The original client was in the Ethernet library and then copied to the Wifi library. Just like I updated the Ethernet version I could just go ahead and apply the same to the Wifi version.